[MHV-51084] Added SM Unit Tests for DeleteDraft and MessageThreads, [MHV-51541] Navigation from unsaved draft - no error message is displayed

vsaleem opened the pull request:
## Summary

- Updated navigation warning to appear when a user navigates away from an unsaved draft
- Enhanced and added new unit tests to cover DeleteDraft and MessageThread components
- All tests are covered at least 81% or above
- Our aim is to achieve unit test coverage of 80% or greater, supporting our overall goal.

## Related issue(s)

### [MHV-51084](https://jira.devops.va.gov/browse/MHV-51084) - SM Unit Tests
>Feature Flag Y/N ? N
>Manual Testing Y/N ? N
>Automated Testing Y/N ? N
>Accessibility Testing Y/N ? N
>
### [MHV-51541](https://jira.devops.va.gov/browse/MHV-51541) - Navigation from unsaved draft - no error message is displayed
>
>Description:
>After making a change to a draft and before saving, then attempting to navigate away from a unsaved draft using left navigation does not show a warning.
>
>User Story:
> As a SM user, I want to be notified if I navigate away from a draft without saving.
>
>GIVEN: After making a change to a draft and before saving,
>WHEN: Attempting to navigate away from unsaved draft using left navigation
>THEN: A warning should display.
>
>Feature Flag Y/N ? N
>Manual Testing Y/N ? Y
>Automated Testing Y/N ? Y
>Accessibility Testing Y/N ? Y
>
>Acceptance Criteria:
>AC1 Error message will be display in scenario above
>AC2 Focus should be on the message once it is displayed

## Testing done

- Unit Tests were added, Manual and Automation tests added
- Overall coverage is at least 90%
- Our aim is to achieve unit test coverage of 80% or greater, supporting our overall goal


## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Before  |<img width="843" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/59583325/ee2c2cca-1744-4137-b691-0307f8279f48">|<img width="787" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/59583325/9b2e0f9f-08ce-45c2-aa99-7b30944cdda6">|
|After |<img width="1316" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/59583325/1e901d82-1a6d-43fd-b9a2-b3450a5bffbf">|   <img width="1324" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/59583325/113396fe-e81b-4c41-bb7c-1c45e0ccdd30">|

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

## Acceptance criteria

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

Commit:
[MHV-47590] Update DeleteDraft Navigation

- user is redirected to previous folder user was in after a unsaved compose/reply draft message is deleted

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
[MHV-47590] Removed former unable to save alert; no longer needed

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge from 'main'

Commit:
[MHV-47590] Updated Delete Draft modal

- added constants for an unsaved new draft / alt. modal

- added props and updated scenarios to populate Delete Draft modal

- Delete Modal appears in Inbox folder

Commit:
[MHV-47590] Cleanup Navigation & DeleteModal Navigation

- Removed NavigationError alert from ReplyForm (this issue will be revisited in the future.)

- Updated navigation calls to function with Delete Draft modal

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
[MHV-47590] Removing cy test for 45118 update

- This updated needed to be removed and will be replanned for a future sprint.  Not totally removing the tests as it will be reintroduced at a later time.

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

fazilqa committed:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

fazilqa committed:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Asterix22 committed:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Asterix22 committed:
adding delete confirmation navigation cypress

Asterix22 committed:
more tests

Commit:
[MHV-47590] Added dispatch successful delete alert after 'deleting' a unsaved draft

Asterix22 committed:
add delete confirmation message verification

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Asterix22 committed:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
[MHV-47590] Cy test update

Commit:
[MHV-47590] Removed unused prop

Commit:
[MHV-47590] Updated prop to delete unsaved, in progress reply draft message

- userSaved prop was removed which was used for this action; added new prop inProgressReplyDraft to check changes to messageBody before saving.

Commit:
[MHV-47590] Cleanup text change

Commit:
[MHV-47590] Fix Navigation Bug in Editable Drafts

- Before unsaved message should receive navigation error if attempting to navigate away

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
Merge branch 'main' into 47590-Delete-Draft-Modal-Update-mhv

Commit:
[MHV-51084] Added SM Unit Tests

- Unit tests covers DeleteDraft and MessageThreads components
- All tests are up to 82% and above
- removed dead tests and updated messageHistory mock
- Created helper function to accept new date format
- cleanup text

KyleCardwell reviewed (approved):
This was a ton of work! Awesome job.

powellkerry committed:
Merge branch 'main' into 51084-Added-SM-Unit-Tests-DeleteDraft-and-MessageThreads

Commit:
[MHV-51084] Skipped flaky tests

Commit:
Merge branch '51084-Added-SM-Unit-Tests-DeleteDraft-and-MessageThreads' of https://github.com/department-of-veterans-affairs/vets-website into 51084-Added-SM-Unit-Tests-DeleteDraft-and-MessageThreads

Commit:
Merge branch 'main' into 51084-Added-SM-Unit-Tests-DeleteDraft-and-MessageThreads

Commit:
[MHV-51541] Navigation Fix for unsaved Drafts

- updated navigation warning to appear if a user navs away from an edited unsaved message

Commit:
cleanup

Commit:
Merge branch 'main' into 51084-Added-SM-Unit-Tests-DeleteDraft-and-MessageThreads

Commit:
[MHV-51084] Removing skipped tests

- skipped tests are causing unit test stability flakiness errors
- removed tests to check stability in gh

Commit:
Merge branch 'main' into 51084-Added-SM-Unit-Tests-DeleteDraft-and-MessageThreads

Commit:
[MHV-51084] Adding Navigation Guard for Reply Draft

- Prevents user from navigating away before a draft is saved

haniasma committed:
Merge branch 'main' into MHV-51541--navigate-away-from-unsaved-draft

Asterix22 committed:
Merge branch 'main' into 51542-navigat-away-unsave-draft

Asterix22 committed:
refactor

Commit:
Merge branch 'main' into 51084-Added-SM-Unit-Tests-DeleteDraft-and-MessageThreads

Commit:
Merge branch 'main' into 51084-Added-SM-Unit-Tests-DeleteDraft-and-MessageThreads

Commit:
Merge branch 'main' into 51084-Added-SM-Unit-Tests-DeleteDraft-and-MessageThreads

Commit:
Merge branch 'main' into 51084-Added-SM-Unit-Tests-DeleteDraft-and-MessageThreads
